tools/verify_stable_archives.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+from urllib.parse import urlparse
+
+from registry import RegistryClient
+
+def verify_stable_archive(url):
+  parsed = urlparse(url)
+  path_parts = parsed.path.split("/")
+  if path_parts[3] != "archive" or path_parts[4] != "refs" or path_parts[5] != "tags":
+    return False
+
+  return True
+
+def main(argv=None):
+  if argv is None:
+    argv = sys.argv[1:]
+
+  client = RegistryClient(".")
+
+  has_failure = False
+  for module_name, version in client.get_all_module_versions():
+    source_url = client.get_source(module_name, version)["url"]
+    if not verify_stable_archive(source_url):
+      has_failure = True
+      print(f'Version `{version}` of module `{module_name}` is using a unstable source url: `{source_url}`')
```

Comment:
nit: "an" unstable source url

Reply:
also please link to somewhere explaining the issue and what you should do to change your URL

Reply:
I added a bit more explanation and a link to the same comment as linked above. If there is another link that better explains both the problem and its resolution, I'd be happy to change it.